### DISCUSSION TITLE
Subresource Integrity - webpack-subresource-integrity has no effect

### DISCUSSION THREAD

**Discussion by af-swilox** [react-static/react-static#1542] [Q&A]
I've added the webpack-subresource-integrity plugin using the webpack API:
```
export default () => ({
    webpack: (config, { stage }) => {
        console.log("----------------------------------------------------------------sri plugin. Stage: "+stage)
        config.output.crossOriginLoading='anonymous'
        config.plugins=[...config.plugins,
            new SriPlugin({hashFuncNames: ["sha256"],enabled:true })
        ]
        return config;
    },
});
```
But neither the `integrity` nor the `crossorigin` attributes are added to the script tags of the generated HTML files. I guess that the index.html files are generated in some specific way that is not supported by the SRI plugin. Any ideas what I can do to make it work?

**Comment by SleeplessByte** [comment]
Hi there! Yes the HTML files are generated using for example [`react-static/src/static/components/HeadWithMeta.js`](https://github.com/react-static/react-static/blob/master/packages/react-static/src/static/components/HeadWithMeta.js). My guess is that the `SriPlugin` runs on `.html` files? 

If it's possible to direct the `SriPlugin` to the _artifacts_ directory, maybe it's still possible.

**Comment by af-swilox** [comment]
Hi @SleeplessByte, I'm not an expert but checking the documentation of the webpack-subresource-integrity plugin, it either works directly with the html-webpack-plugin and then everything should just work. If the html-webpack-plugin is not used, then the integrity and crossorigin attributes need to be set by the html generator and e.g. the integrity values can be accessed from the webpack model. This snippet is taken directly from the webpack-subresource-integrity plugin documentation:
```
compiler.plugin("done", stats => {
    const mainAssetName = stats.toJson().assetsByChunkName.main;
    const integrity = stats.compilation.assets[mainAssetName].integrity;
});
```
Can you give me some more input on how the html files are generated? I guess webpack does the packaging of the .js files (main, vendor, ...) and e.g. add the hash to the name. Is the html-webpack-plugin used for that?
Thanks!
